### PR TITLE
Modifies eig_p and related operations to take advantage of the new jaxlib geev API

### PR DESCRIPTION
**Pull request by bchetioui**
This follows from #3882. @hawkinsp's comments on that PR have been addressed here.

### CODE REVIEW

File: build/test-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 flake8
-jaxlib==0.1.51
+jaxlib==0.1.55
```
Comment (skye) [inline build/test-requirements.txt:2]: You should also bump the minimum jaxlib here: https://github.com/google/jax/blob/master/jax/lib/__init__.py#L25
(In fact, can you also add a comment to the line here saying to always keep these versions in sync?)

**Review by skye — CHANGES_REQUESTED** [review]

File: build/test-requirements.txt (hunk 1 above)
Comment (SIben) [reply build/test-requirements.txt]: Done, thank you!

File: build/test-requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 flake8
-jaxlib==0.1.51
+jaxlib==0.1.55 # must be kept in sync with the version in jax/lib/__init__.py
```
Comment (skye) [inline build/test-requirements.txt:2]:
```suggestion
 # Must be kept in sync with the minimum jaxlib version in jax/lib/__init__.py
jaxlib==0.1.55
```

**Review by skye — APPROVED** [review]
One tiny nit, otherwise LGTM!

File: build/test-requirements.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 flake8
-jaxlib==0.1.51
+ # Must be kept in sync with the minimum jaxlib version in jax/lib/__init__.py
```
Comment (gnecula) [inline build/test-requirements.txt:2]: Isn't the minimum version 1.51 still? What I would do is to bump the minimum version (and the version here) in a few days, unless we discover a problem with 1.55

Comment (gnecula) [reply build/test-requirements.txt:2]: I see you've done it below. Ok.

**Pull request merged**